pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -497,7 +493,9 @@ private void sendOutputMessage(InputMessage srcMsg,
                                    JavaExecutionResult result,
                                    byte[] output) {
         MessageBuilder msgBuilder = MessageBuilder.create()
-            .setContent(output);
+            .setContent(output)
+            .setProperty("__pfn_input_topic__", srcMsg.getTopicName())
+            .setProperty("__pfn_msg_id__", new String(Base64.getEncoder().encode(srcMsg.getActualMessage().getMessageId().toByteArray())));
```

Comment:
Should this be more explicit it refers to the source message? 
Like `__pfn_src_msg_id__` or `__pfn_input_msg_id__

Reply:
Changed